src/atoms/affine/sum.jl

```diff
@@ -57,9 +57,13 @@ function conic_form!(x::SumAtom, unique_conic_forms::UniqueConicForms=UniqueConi
   return get_conic_form(unique_conic_forms, x)
 end
 
-sum(x::AbstractExpr) = SumAtom(x)
+# Dispatch to an internal helper function that handles the dimension argument in
+# the same manner as Base, with dims=: denoting a regular sum
+sum(x::AbstractExpr; dims=:) = _sum(x, dims)
```

Comment:
Should we put the comment above as docstring for `sum(::AbstractExpr)`?

Reply:
I don't think so, since the fact that this dispatches to a helper function which isn't exported is an implementation detail.

Reply:
That no indeed, but the
```julia
handles the dimension argument in
the same manner as Base, with dims=: denoting a regular sum
```
Could be a docstring

Reply:
Yeah true. I'd rather make documentation-related changes and other cleanup tasks in a separate PR though.

Reply:
In that case the comment lines are added in the PR, so it wouldn't be out of scope to make it a docstring right away